mathics/builtin/datentime.py

```diff
@@ -1134,27 +1134,11 @@ class TimeConstrained(Builtin):
 
     def eval_2(self, expr, t, evaluation) -> Optional[BaseElement]:
         "TimeConstrained[expr_, t_]"
-        t = t.evaluate(evaluation)
-        if not t.is_numeric(evaluation):
-            evaluation.message("TimeConstrained", "timc", t)
-            return
-        try:
-            timeout = float(t.to_python())
-        except (ValueError, TypeError):
-            return
-        return eval_timeconstrained(expr, timeout, SymbolAborted, evaluation)
+        return eval_timeconstrained(expr, t, SymbolAborted, evaluation)
```

Comment:
To be clear about what goes into the evaluation _method_ of Builtin object class versus the evaluation interpreter function. 

It is okay to do parameter type checking and parameter conversion in the class method, especially when interactions inside an interpreter are not likely to need to do this kind of thing. 

There is a bit of wiggle room for where to do this, so use your judgment.